DEVELOP.md

```diff
@@ -0,0 +1,485 @@
+specification.  IBM DFDL lists some more [unsupported
+features](https://www.ibm.com/docs/en/integration-bus/10.0?topic=dfdl-unsupported-features)
+and also lists some [implementation-specific
+limits](https://www.ibm.com/docs/en/integration-bus/10.0?topic=dfdl-implementation-specific-limits).
+These limitations will not prevent you from writing DFDL schemas for
+almost all data formats, but they will reveal which parts of the DFDL
+specification are rarely used.
+
+## Daffodil Development
+
+The Apache Software Foundation hosts the Apache Daffodil project on
+the following ASF infrastructure:
+
+- Daffodil's [issue
+  tracker](https://issues.apache.org/jira/projects/DAFFODIL/) is
+  hosted on JIRA
+- Daffodil's
+  [users](https://lists.apache.org/list.html?users@daffodil.apache.org),
+  [dev](https://lists.apache.org/list.html?dev@daffodil.apache.org),
+  and
+  [commits](https://lists.apache.org/list.html?commits@daffodil.apache.org)
+  mailing lists are hosted on Apache Pony Mail
+- Daffodil's [source code](https://github.com/apache/daffodil) is
+  hosted on GitHub
+- Daffodil's
+  [wiki](https://cwiki.apache.org/confluence/display/DAFFODIL/) is
+  hosted on Confluence
+- Daffodil's [website](https://daffodil.apache.org/) is hosted on
+  Apache with static content generated by Jekyll using jekyll-asciidoc
+  and asciidoctor-diagram plugins
+
+A good Daffodil developer has a GitHub account and two copies of
+Daffodil's source code repository, a reading copy directly cloned from
+the Apache repository and a working copy cloned from the developer's
+own fork of the Apache repository.  A good Daffodil developer also
+gets an account on the Apache Software Foundation's JIRA and
+Confluence servers and subscribes to all three Daffodil mailing lists.
+
+You can clone your reading copy directly from the Apache repository,
+but you should never make any changes in it.  The only commands you
+should ever run in your reading copy are `git pull`, `git log
+ORIG_HEAD..HEAD`, and `git diff ORIG_HEAD..HEAD` in order to see what
+changes have been made by other developers since your last pull from
+the Apache repository.  Your reading copy should remain an exact copy
+of the Apache source code repository at all times which you can use
+for reading or occasionally running `diff` commands between your
+reading copy and working copy.  To be safe, rename your reading copy
+from `daffodil` to `daffodil-asf` and never edit files or run `sbt` in
+it so there will be nothing to push to the Apache repository even if
+you accidentally run `git push` from the wrong source tree.
+
+You can set up your working copy by forking the Apache repository to
+your own GitHub account and then cloning your working copy from your
+fork.  Cloning your working copy from your fork instead of the Apache
+repository ensures that any changes you commit in your working copy
+will be pushed safely to only your own fork, not the Apache
+repository.
```

Comment:
I might suggest just replacing this section with a link to the contribution workflow page? That provides steps git commands to set up a repo, which I think is easier to follow than this written prose.

I also am not sure I totally agree with the reading vs writing copy. For new contributors, there's not really a difference because they don't have write access to apache/daffodil. For them, I'd just recommend a single repo with two remotes (which is what the instructions recommend).

We may even want to create a CONTRIBUTING.md file that moves the "Contributor Workflow" from he wiki to github so it's a bit more visible.